src/compiler/emitter.ts

```diff
@@ -3054,7 +3054,31 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                     node.parent.kind === SyntaxKind.SourceFile;
             }
 
+            function filterNonInitializedExportedVariableDeclarations(nodes: NodeArray<VariableDeclaration>, flag: NodeFlags): NodeArray<VariableDeclaration> {
+                let result = <NodeArray<VariableDeclaration>>Object.create(nodes);
+                result.splice(0);
+                for (let i = 0; i<nodes.length; i++) {
+                    if (!(nodes[i].parent.parent.flags & NodeFlags.Export && !nodes[i].initializer)) {
+                        result.push(nodes[i]);
+                    }
+                }
+                return result;
+            }
+
             function emitVariableStatement(node: VariableStatement) {
+                let variableDeclarations: NodeArray<VariableDeclaration>;

```

Comment:
But nobody was using `pos` and `end` and you weren't setting it anyway :smile: 

Notice that emitCommaList only takes a `Node[]`.